lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,28 @@
+module Semian
+  class AtomicInteger #:nodoc:
+    attr_accessor :value
+
+    def initialize(_name, _permissions)
+      @value = 0
+    end
+
+    def increment_by(val)
+      self.value += val
+    end
+
+    def increment
+      increment_by(1)
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
+
+    def shared?
+      false
+    end
```

Comment:
So what's the consensus for `#shared?`? @byroot @Sirupsen. The most important reason is outlined above:
> Yes, essentially I want something that lets me differentiate between a structure that has shared memory (and therefore needs a different set of methods for `destroy` (since we now need to free the semaphore, etc), `execute_atomically` (which locks, calls yield, and unlocks). However, because we now have a fallback implementation as well, this functionality can't just be a complete override, so we need some `bool` of some sort to determine whether we are actually using shared memory and branch off. Thats what the `#shared?` was for.

If `#shared?` isn't included (it could be `private` for all I care) then I need a different way of implementing falbacks.